### PR TITLE
gRPC template updates

### PR DESCRIPTION
**Pull request by JunTaoLuo**
Addresses https://github.com/grpc/grpc-dotnet/issues/141 and https://github.com/aspnet/AspNetCore/issues/10820

### CODE REVIEW

**Comment by JunTaoLuo** [comment]
I'm able to test the project template in VS and command line. I've tested the item template on the command line but I don't think there's a good way to test it in vs cc @BillHiebert @phenning ?

File: src/ProjectTemplates/Web.ProjectTemplates/content/GrpcService-CSharp/Properties/launchSettings.json [hunk 1]
```diff
@@ -3,7 +3,7 @@
     "GrpcService-CSharp": {
       "commandName": "Project",
       "launchBrowser": false,
-      "applicationUrl": "http://localhost:50051",
+      "applicationUrl": "https://localhost:50051",
```
Comment (shirhatti) [inline src/ProjectTemplates/Web.ProjectTemplates/content/GrpcService-CSharp/Properties/launchSettings.json]: Do we want to continue to use a custom port? My vote would be against it

**Comment by shirhatti** [comment]
@JunTaoLuo Can you confirm this blows up on a Mac as expected by https://github.com/aspnet/AspNetCore/issues/11061?

File: src/ProjectTemplates/Web.ProjectTemplates/content/GrpcService-CSharp/Properties/launchSettings.json (hunk 1 above)
Comment (JunTaoLuo) [reply src/ProjectTemplates/Web.ProjectTemplates/content/GrpcService-CSharp/Properties/launchSettings.json]: So just https://localhost:5001 then?

Comment (shirhatti) [reply src/ProjectTemplates/Web.ProjectTemplates/content/GrpcService-CSharp/Properties/launchSettings.json]: Yes

**Comment by JunTaoLuo** [comment]
Error at startup:

```
warn: Microsoft.AspNetCore.Server.Kestrel[0]
      Unable to bind to https://localhost:5001 on the IPv4 loopback interface: 'HTTP/2 over TLS is not supported on OSX due to missing ALPN support.'.
warn: Microsoft.AspNetCore.Server.Kestrel[0]
      Unable to bind to https://localhost:5001 on the IPv6 loopback interface: 'HTTP/2 over TLS is not supported on OSX due to missing ALPN support.'.
crit: Microsoft.AspNetCore.Server.Kestrel[0]
      Unable to start Kestrel.
System.IO.IOException: Failed to bind to address https://localhost:5001.
 ---> System.AggregateException: One or more errors occurred. (HTTP/2 over TLS is not supported on OSX due to missing ALPN support.) (HTTP/2 over TLS is not supported on OSX due to missing ALPN support.)
 ---> System.NotSupportedException: HTTP/2 over TLS is not supported on OSX due to missing ALPN support.
   at Microsoft.AspNetCore.Server.Kestrel.Https.Internal.HttpsConnectionMiddleware..ctor(ConnectionDelegate next, HttpsConnectionAdapterOptions options, ILoggerFactory loggerFactory)
   at Microsoft.AspNetCore.Hosting.ListenOptionsHttpsExtensions.<>c__DisplayClass12_0.<UseHttps>b__0(ConnectionDelegate next)
   at Microsoft.AspNetCore.Server.Kestrel.Core.ListenOptions.Build()
   at Microsoft.AspNetCore.Server.Kestrel.Core.KestrelServer.<>c__DisplayClass21_0`1.<<StartAsync>g__OnBind|0>d.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.AddressBinder.BindEndpointAsync(ListenOptions endpoint, AddressBindContext context)
   at Microsoft.AspNetCore.Server.Kestrel.Core.LocalhostListenOptions.BindAsync(AddressBindContext context)
   --- End of inner exception stack trace ---
 ---> (Inner Exception #1) System.NotSupportedException: HTTP/2 over TLS is not supported on OSX due to missing ALPN support.
   at Microsoft.AspNetCore.Server.Kestrel.Https.Internal.HttpsConnectionMiddleware..ctor(ConnectionDelegate next, HttpsConnectionAdapterOptions options, ILoggerFactory loggerFactory)
   at Microsoft.AspNetCore.Hosting.ListenOptionsHttpsExtensions.<>c__DisplayClass12_0.<UseHttps>b__0(ConnectionDelegate next)
   at Microsoft.AspNetCore.Server.Kestrel.Core.ListenOptions.Build()
   at Microsoft.AspNetCore.Server.Kestrel.Core.KestrelServer.<>c__DisplayClass21_0`1.<<StartAsync>g__OnBind|0>d.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.AddressBinder.BindEndpointAsync(ListenOptions endpoint, AddressBindContext context)
   at Microsoft.AspNetCore.Server.Kestrel.Core.LocalhostListenOptions.BindAsync(AddressBindContext context)<---
```

**Comment by JunTaoLuo** [comment]
😮 We still call it OSX instead of macOS?

File: src/ProjectTemplates/Web.ProjectTemplates/content/GrpcService-CSharp/Properties/launchSettings.json (hunk 1 above)
Comment (JunTaoLuo) [inline src/ProjectTemplates/Web.ProjectTemplates/content/GrpcService-CSharp/Properties/launchSettings.json]:
```suggestion
      "applicationUrl": "https://localhost:5001",
```

**Comment by shirhatti** [comment]
> We still call it OSX instead of macOS?

Looks like we aren't consistent. We're all over the place on this 😞

**Review by shirhatti — APPROVED** [review]

**Comment by shirhatti** [comment]
@DamianEdwards FYI. Based on our conversation earlier, the gRPC templates will use HTTPS by default. It will blow up on macOS on Startup where there is no server-side ALPN.

File: src/ProjectTemplates/Web.ProjectTemplates/content/GrpcService-CSharp/Properties/launchSettings.json (hunk 1 above)
Comment (JamesNK) [reply src/ProjectTemplates/Web.ProjectTemplates/content/GrpcService-CSharp/Properties/launchSettings.json]: When you talk to Jan next please ask him if there is any reason that gRPC demo apps have ports in 5005x range.

**Review by JamesNK — APPROVED** [review]
I'd like to double check with Jan about why gRPC samples use ports in 5005x.

Also need to update the linked tutorial to add HTTPS, and change the port used in it. Shouldn't be merged until preview 7 ships

File: eng/Versions.props [hunk 2]
```diff
@@ -216,9 +216,7 @@
     <BenchmarkDotNetPackageVersion>0.10.13</BenchmarkDotNetPackageVersion>
     <CastleCorePackageVersion>4.2.1</CastleCorePackageVersion>
     <FSharpCorePackageVersion>4.2.1</FSharpCorePackageVersion>
-    <GoogleProtobufPackageVersion>3.8.0</GoogleProtobufPackageVersion>
-    <GrpcAspNetCoreServerPackageVersion>0.1.21-pre1</GrpcAspNetCoreServerPackageVersion>
-    <GrpcToolsPackageVersion>1.21.0</GrpcToolsPackageVersion>
+    <GrpcAspNetCorePackageVersion>0.1.22-pre1</GrpcAspNetCorePackageVersion>
```
Comment (JunTaoLuo) [inline eng/Versions.props]: Note. Need to publish a version of this metapackage.

**Comment by ryanbrandenburg** [comment]
```
##[error]src\SignalR\samples\SocialWeather\SocialWeather.csproj(0,0): error NU1604: Project dependency Google.Protobuf does not contain an inclusive lower bound. Include a lower bound in the dependency version to ensure consistent restore results.
```
Looks like this broke the SignalR samples because it removed the Google.Protobuf version.

**Comment by JunTaoLuo** [comment]
Ahh... I thought we were the only ones using it.

**Comment by ryanbrandenburg** [comment]
> Ahh... I thought we were the only ones using it.

It's good practice to do a "Ctrl-F" of the whole repo for references before removing anything (class, package, etc). Even if it SHOULDN'T be used by something else sometimes it is anyway.

File: eng/Versions.props (hunk 2 above)
Comment (JunTaoLuo) [reply eng/Versions.props]: We'll update this to 0.1.22-pre2 when it gets published. this version is currently published to the nightly feed so we this PR is unblocked.

**Pull request merged**